libpod/container_internal_linux.go

```diff
@@ -368,6 +368,29 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 		g.AddMount(overlayMount)
 	}
 
+	// Add image volumes as overlay mounts
+	for _, volume := range c.config.ImageVolumes {
+		// Mount the specified image.
+		img, err := c.runtime.ImageRuntime().NewFromLocal(volume.Source)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error creating image volume %q:%q", volume.Source, volume.Dest)
+		}
+		mountPoint, err := img.Mount(nil, "")
```

Comment:
Not sure if SELinux will block the image without a label. But need to verify this will work. On the other hand this had better work, since the mounts will be shared between different containers.

Reply:
Need to handle read/only mounts